### PR TITLE
Add "Add the Spatial Mapping Prefab" to the scene configuration window

### LINKED ISSUE #2203: Proposal:We were wondering if you could add the checkbox:"Add the Spatial Mapping prefab" in "Apply Mixed Reality Scene Setting" dialog?
## Overview

"Apply Mixed Reality Project Setting"  has useful to add "Set Default Spatial Mapping Layer",but we might misunderstand that "Spatial Mapping" is become to available.
In actuality,we must set "SpatialMapping.prefab" to the scene, so we ware wonderring if you could add the checkbox:"Add the Spatial Mapping prefab" in "Apply Mixed Reality Scene Setting" dialog?

image

we can set both project and scene from configuration dialog,this should come in useful.

## Unity Editor Version
Unity 2017.4.3f1

## Mixed Reality Toolkit Release Version
MRTK 2017.4.0.0-rc1

### PR DESCRIPTION
**Pull request by keveleigh**
Overview
---
Adds an optional setting to the scene configuration window to quickly add the SpatialMapping prefab to the scene.

Changes
---
- Fixes #2203

### CODE REVIEW

File: Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs [hunk 1]
```diff
@@ -129,6 +137,17 @@ protected override void ApplySettings()
                 FindObjectOfType<InputManager>().GetComponent<SimpleSinglePointerSelector>().Cursor = FindObjectOfType<Cursor>();
             }
 
+            if (Values[SceneSetting.AddSpatialMapping])
+            {
+                var spatialMappingManagers = FindObjectsOfType<SpatialMappingManager>();
+                foreach (var spatialMappingManager in spatialMappingManagers)
+                {
+                    DestroyImmediate(spatialMappingManager.gameObject.GetParentRoot());
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: Do you want to destroy the GameObject?

What if there's other components ppl are using on it?

Comment (davidkline-ms) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: destroying the parent seems risky.

should we just leave what we find alone if it

also, we should warn users if we find more than one pre-existing spatial mapping manager

Comment (keveleigh) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: Ahh yeah, I copied the line above which destroys the cursor's parent: https://github.com/keveleigh/HoloToolkit-Unity/blob/9525e0803aacc26dfa13b493a0f44300c656137d/Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs#L132

I can update it to instead just destroy its own GameObject, like the InputSystem does (multiple times): https://github.com/keveleigh/HoloToolkit-Unity/blob/9525e0803aacc26dfa13b493a0f44300c656137d/Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs#L92

or should all of this behavior be updated?

Comment (davidkline-ms) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/SceneSettingsWindow.cs]: I'd rather have a warning of too many spatial mapping managers and leave the existing one(s) untouched

**Pull request merged**